src/extract.rs

```diff
@@ -267,6 +267,43 @@ mod test {
         assert_eq!(links, HashSet::new())
     }
 
+    #[test]
+    fn test_markdown_internal_url() {
+        let input = "This is [an internal url](@/internal.md).";
+        let links: HashSet<Uri> = match Url::parse("https://localhost.com/") {
```

Comment:
Why do a `match` statement here?
It might be needed but I don't get it. 😄 At least in the test below (`test_skip_markdown_email`) you didn't have a `match` because you knew the output already. Was expecting something like this here, too.
If you are worried about the `Result` type, you can just `unwrap()` it here. Since we are in a unit test that's fine as `unwrap()` will merely panic, which is what will be recognized as a failed test.

Reply:
Will fix this!